File: backend/src/main/java/org/pupille/backend/mysql/screening/TerminDTOWithFilmDTOOverviews.java
package org.pupille.backend.mysql.screening;

import org.pupille.backend.mysql.film.Film;
import org.pupille.backend.mysql.termin.Termin;

import java.time.LocalDateTime;
import java.util.List;

public record TerminDTOWithFilmDTOOverviews(
        Long terminId,
        LocalDateTime screeningTime,
        String titel,
        String kurztext,
        String besonderheit,
        Integer sonderfarbe,
        List<FilmDTOOverview> films
) {
    public TerminDTOWithFilmDTOOverviews(Termin termin, List<Film> films) {
        this(
                termin.getTnr(),
                termin.getTermin(),
                termin.getTitel(),
                termin.getKurztext(),
                termin.getBesonderheit(),
                termin.getSonderfarbe(),
                films.stream()
                        .map(FilmDTOOverview::new)
                        .toList()
        );
    }
}

File: frontend/src/types/TerminDTOWithFilmDTOOverviews.ts
import FilmDTOOverview from "./FilmDTOOverview.ts";

// this one mirrors TerminDTOWithFilmDTOOverviews
interface TerminDTOWithFilmDTOOverviews {
    terminId: number;
    screeningTime?: string;
    titel?: string | null;
    kurztext?: string | null;
    besonderheit?: string| null;
    bild?: string| null;
    sonderfarbe?: number | null;
    veroeffentlichen?: number | null;
    films: FilmDTOOverview[];
}

export default TerminDTOWithFilmDTOOverviews;